h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -270,8 +276,10 @@ public static ParseSetup guessSetup( Key[] fkeys, ParseSetup userSetup ) {
     if (ice instanceof Frame && ((Frame) ice).vec(0) instanceof UploadFileVec) {
       t._gblSetup._chunk_size = FileVec.DFLT_CHUNK_SIZE;
     } else {
-      t._gblSetup._chunk_size = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
-              Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int calculatedChunkSize = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
+          Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int finalChunkSize = Math.max(t._gblSetup.dataOffset + 2, calculatedChunkSize);
```

Comment:
This is a crutch, I understand.
Next I'm planning to slightly fix the parsers so they'll be able to deal with headers of any length, per file, or global, and not split input into chunks before carefully looking at it.
This current pr just solves one specific arrf problem.
I tried to move dataOffset to arrf, it's natural, but it did not work. The calculation of "chunk size" is happening where it is happening, and arrf parser has nothing to do with it, unfortunately. Changing it would mean redesigning the parsers. I hope to do it next.